src/graphics/rect.rs

```diff
@@ -22,6 +26,267 @@ pub type IntRect = Rect<i32>;
 /// A [`Rect`] of `f32`.
 pub type FloatRect = Rect<f32>;
 
+pub trait LossyFrom<T> {
+    fn lossy_from(v: T) -> Self;
```

Comment:
I don't think we need traits for this. We could just have inherent methods called `as_f32`, `as_f64`, `as_i8`, etc.
That way we don't pollute the API with traits.

Also it looks like these traits are pasted into each module (rect, vector2, vector3), which is definitely not ideal.